pkg/helm/watcher/controller/helm_watcher.go

```diff
@@ -140,3 +159,82 @@ func (r *HelmWatcherReconciler) reconcileDelete(ctx context.Context, repository
 	// Stop reconciliation as the object is being deleted
 	return ctrl.Result{}, nil
 }
+
+// sendEvent emits a Kubernetes event and forwards the event to notification controller if configured.
```

Comment:
I don't see where in this code it is emitting a Kubernetes event?

Reply:
Ops, good spot